<a href="https://colab.research.google.com/github/Matthew-Calvario/github-introfall25-Matthew-Calvario/blob/main/lab_4_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import seaborn as sns
iris = sns.load_dataset('iris')


In [3]:
virginica_sepal_width = iris[iris['species'] == 'virginica']['sepal_width']
min_sepal_width_virginica = virginica_sepal_width.min()
print(f"The minimum sepal_width for the 'virginica' variety is: {min_sepal_width_virginica}")

The minimum sepal_width for the 'virginica' variety is: 2.2


In [6]:
mean_petal_length = iris['petal_length'].mean()
filtered_iris = iris[iris['petal_length'] > mean_petal_length]
display(f"The mean petal_length of the entire iris dataset is: {mean_petal_length}")
display(filtered_iris.head())

'The mean petal_length of the entire iris dataset is: 3.7580000000000005'

sepal_length  sepal_width  petal_length  petal_width     species
50           7.0          3.2           4.7          1.4  versicolor
51           6.4          3.2           4.5          1.5  versicolor
52           6.9          3.1           4.9          1.5  versicolor
53           5.5          2.3           4.0          1.3  versicolor
54           6.5          2.8           4.6          1.5  versicolor

In [7]:
num_observations = len(filtered_iris)
display(f"The number of observations in the new DataFrame (where petal_length > mean_petal_length) is: {num_observations}")

'The number of observations in the new DataFrame (where petal_length > mean_petal_length) is: 93'

In [10]:
correlation_sepal_length_sepal_width = iris['sepal_length'].corr(iris['sepal_width'])
print(f"Pearson correlation between sepal_length and sepal_width: {correlation_sepal_length_sepal_width:.4f}")

Pearson correlation between sepal_length and sepal_width: -0.1176


In [11]:
correlation_petal_length_sepal_length = iris['petal_length'].corr(iris['sepal_length'])
print(f"Pearson correlation between petal_length and sepal_length: {correlation_petal_length_sepal_length:.4f}")

Pearson correlation between petal_length and sepal_length: 0.8718


In [12]:
correlation_sepal_length_petal_width = iris['sepal_length'].corr(iris['petal_width'])
print(f"Pearson correlation between sepal_length and petal_width: {correlation_sepal_length_petal_width:.4f}")

Pearson correlation between sepal_length and petal_width: 0.8179


In [13]:
correlation_petal_length_petal_width = iris['petal_length'].corr(iris['petal_width'])
print(f"Pearson correlation between petal_length and petal_width: {correlation_petal_length_petal_width:.4f}")

Pearson correlation between petal_length and petal_width: 0.9629


In [15]:
average_sepal_width_by_species = iris.groupby('species')['sepal_width'].mean()
display(f"Average sepal_width by species:\n{average_sepal_width_by_species}")

'Average sepal_width by species:\nspecies\nsetosa        3.428\nversicolor    2.770\nvirginica     2.974\nName: sepal_width, dtype: float64'

In [14]:
# Mean petal_length of setosa
setosa_petal_length_mean = iris[iris['species'] == 'setosa']['petal_length'].mean()
display(f"Mean petal_length of setosa: {setosa_petal_length_mean:.4f}")

# Iris with the largest sepal_width
largest_sepal_width_iris = iris.loc[iris['sepal_width'].idxmax()]
display(f"Iris with the largest sepal_width:\n{largest_sepal_width_iris}")

# Standard deviation of sepal_length of setosa
setosa_sepal_length_std = iris[iris['species'] == 'setosa']['sepal_length'].std()
display(f"Standard deviation of sepal_length of setosa: {setosa_sepal_length_std:.4f}")

# Iris with the largest sepal_length
largest_sepal_length_iris = iris.loc[iris['sepal_length'].idxmax()]
display(f"Iris with the largest sepal_length:\n{largest_sepal_length_iris}")

'Mean petal_length of setosa: 1.4620'

'Iris with the largest sepal_width:\nsepal_length       5.7\nsepal_width        4.4\npetal_length       1.5\npetal_width        0.4\nspecies         setosa\nName: 15, dtype: object'

'Standard deviation of sepal_length of setosa: 0.3525'

'Iris with the largest sepal_length:\nsepal_length          7.9\nsepal_width           3.8\npetal_length          6.4\npetal_width           2.0\nspecies         virginica\nName: 131, dtype: object'

In [9]:
mean_sepal_length_rounded = iris['sepal_length'].mean()
num_unique_species = iris['species'].nunique()
num_virginica = len(iris[iris['species'] == 'virginica'])

display(f"Mean sepal_length (rounded to 3 digits): {mean_sepal_length_rounded:.3f}")
display(f"Number of unique species: {num_unique_species}")
display(f"Number of 'virginica' species: {num_virginica}")

'Mean sepal_length (rounded to 3 digits): 5.843'

'Number of unique species: 3'

"Number of 'virginica' species: 50"

In [8]:
num_versicolor = len(iris[iris['species'] == 'versicolor'])
display(f"The number of observations for 'versicolor' species is: {num_versicolor}")

"The number of observations for 'versicolor' species is: 50"

In [16]:
import seaborn as sns
titanic = sns.load_dataset('titanic')

In [20]:
print('Missing values in Iris dataset:')
missing_iris = iris.isnull().sum()
display(missing_iris[missing_iris > 0])

print('\nMissing values in Titanic dataset:')
missing_titanic = titanic.isnull().sum()
display(missing_titanic[missing_titanic > 0])

Missing values in Iris dataset:


Series([], dtype: int64)


Missing values in Titanic dataset:


age            177
embarked         2
deck           688
embark_town      2
dtype: int64

In [18]:
first_class_not_survived_ages = titanic[(titanic['pclass'] == 1) & (titanic['survived'] == 0)]['age']
min_age_first_class_not_survived = first_class_not_survived_ages.min()
max_age_first_class_not_survived = first_class_not_survived_ages.max()

display(f"The minimum age for 1st class passengers who did not survive is: {min_age_first_class_not_survived}")
display(f"The maximum age for 1st class passengers who did not survive is: {max_age_first_class_not_survived}")

'The minimum age for 1st class passengers who did not survive is: 2.0'

'The maximum age for 1st class passengers who did not survive is: 71.0'

In [19]:
female_passengers = titanic[titanic['sex'] == 'female']
num_female_survived = female_passengers['survived'].sum()
total_female_passengers = len(female_passengers)

if total_female_passengers > 0:
    percentage_female_survived = (num_female_survived / total_female_passengers) * 100
    rounded_percentage = round(percentage_female_survived)
    display(f"The percentage of female passengers who survived is: {rounded_percentage:.0f}%")
else:
    display("There are no female passengers in the dataset.")

'The percentage of female passengers who survived is: 74%'

In [17]:
survived_second_class_passengers = titanic[(titanic['pclass'] == 2) & (titanic['survived'] == 1)]
num_survived_second_class = len(survived_second_class_passengers)
display(f"The number of passengers in the 2nd class who survived is: {num_survived_second_class}")

'The number of passengers in the 2nd class who survived is: 87'

# Task
Calculate the correlation between 'survived' and 'pclass' and between 'survived' and 'fare' in the `titanic` dataset, then summarize whether the given statements are true or false based on all the analyses performed so far.

## Correlation with Passenger Class

### Subtask:
Calculate the correlation between 'survived' and 'pclass' to verify the fourth statement. Note that a lower 'pclass' value indicates a higher class.


**Reasoning**:
To calculate the correlation between 'survived' and 'pclass', I will use the `.corr()` method on the `titanic` DataFrame.



In [24]:
correlation_survived_pclass = titanic['survived'].corr(titanic['pclass'])
print(f"Pearson correlation between 'survived' and 'pclass': {correlation_survived_pclass:.4f}")

Pearson correlation between 'survived' and 'pclass': -0.3385


## Correlation with Fare

### Subtask:
Calculate the correlation between 'survived' and 'fare' to further analyze factors affecting survival rates.

**Reasoning**:
To calculate the correlation between 'survived' and 'fare', I will use the `.corr()` method on the `titanic` DataFrame.



In [25]:
correlation_survived_fare = titanic['survived'].corr(titanic['fare'])
print(f"Pearson correlation between 'survived' and 'fare': {correlation_survived_fare:.4f}")

Pearson correlation between 'survived' and 'fare': 0.2573


## Final Task

### Subtask:
Summarize the findings for each statement, indicating whether it is true or false based on the analysis.


## Summary:

### Data Analysis Key Findings

*   There is a negative Pearson correlation of -0.3385 between 'survived' and 'pclass', indicating that passengers in lower numerical classes (higher actual classes) had a better chance of survival.
*   There is a positive Pearson correlation of 0.2573 between 'survived' and 'fare', suggesting that passengers who paid a higher fare had a somewhat better chance of survival.

### Insights or Next Steps

*   Both passenger class and fare are correlated with survival, with higher classes and higher fares generally linked to increased survival rates. This aligns with the understanding that socio-economic status likely played a role in survival outcomes during the Titanic disaster.
